sklearn/preprocessing/_discretization.py

```diff
@@ -279,9 +284,9 @@ def transform(self, X):
         np.clip(Xt, 0, self.n_bins_ - 1, out=Xt)
 
         if self.encode == 'ordinal':
-            return Xt
+            return Xt.astype(self.dtype)
```

Comment:
Similarly let's use `_astype_copy_false` helper, as mentioned before.

Reply:
I don't have `astype` anymore in the second solution.